bin/buildBlueprints.js

```diff
@@ -111,6 +112,18 @@ if (argv.watch) {
   extensions.watch = true;
 }
 
+function addCacheBustedFile(name, m) {
+  var filePath = './.test/' + name;
+  return new Promise(function(resolve, reject) {
+    var cacheBuster = (new Date()).getTime();
+    var newPath = filePath + '-' + cacheBuster;
+    fs.copy(path.resolve(filePath), path.resolve(newPath), function() {
+      m.addFile(newPath);
```

Comment:
While I agree with Niranjan - just make this synchronous and keep it simple - as a general design guideline, I don't think you should have another side effect in here. Promises are meant as a remedy to that. Can you just `resolve(newPath)` here, then capture that resolutions in the `then` callback.